### PR TITLE
Fix: Invalid padding calculation in FuseConvAndPoolPadding PatternCallback

### PR DESCRIPTION
**Pull request by ashokkumarkannan1**
Fix https://github.com/tenstorrent/tt-forge-fe/issues/520

### CODE REVIEW

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 1]
```diff
@@ -147,13 +147,23 @@ def callback(self, pre, post, node_map):
...
            return
 
         pad_width = pad.attrs.pad_width
-        padding = list(pad_width[-2]) + list(pad_width[-3]) # left, right, top, bottom
+        top = pad_width[-2][0]
+        bottom = pad_width[-2][1]
+        left = pad_width[-1][0]
+        right = pad_width[-1][1]
+
+        assert top == bottom, "padding top and padding bottom must be same"
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:162]: Add brief more details about why these must be the same. Also, if this is a software limitation, let's add a link to the issue that will remove this constrain in the future.

Comment (ashokkumarkannan1) [reply python/tvm/relay/op/contrib/forge/forge_passes.py:162]: TTNN doesn't support different padding values. I will add the comments accordingly.

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 2]
```diff
@@ -147,13 +147,26 @@ def callback(self, pre, post, node_map):
...
         pad = node_map[self.pad][0]
+
         if not all(conv_pool.attrs.padding[i] == 0 for i in range(len(conv_pool.attrs.padding))) \
            or not isinstance(pad.args[1], tvm.relay.Constant) or not pad.args[1].data.shape == () \
            or not int(pad.args[1].data.numpy()) == 0:
            return
 
         pad_width = pad.attrs.pad_width
-        padding = list(pad_width[-2]) + list(pad_width[-3]) # left, right, top, bottom
+        top = pad_width[-2][0]
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:157]: Let's rename those to include `_pad` suffix. E.g. `top_pad`

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 3]
```diff
@@ -147,13 +147,26 @@ def callback(self, pre, post, node_map):
...
+        bottom = pad_width[-2][1]
+        left = pad_width[-1][0]
+        right = pad_width[-1][1]
+
+        # Different padding values currently not supported in TTNN
+        # For Conv2dOp Op, refer third_party/tt-mlir/lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:578
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:163]: Let's refer to file path and function name, as line with often change :))

File: python/tvm/relay/op/contrib/forge/forge_passes.py [hunk 4]
```diff
@@ -147,13 +147,26 @@ def callback(self, pre, post, node_map):
...
+        # For Maxpool2d Op, refer third_party/tt-mlir/lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp:614
+        assert top == bottom, "padding top and padding bottom must be same"
```
Comment (nvukobratTT) [inline python/tvm/relay/op/contrib/forge/forge_passes.py:165]: Let's assert with the following message: "Unsupported different top and bottom padding size". Let's set a similar msg to the second assert as well

File: python/tvm/relay/op/contrib/forge/forge_passes.py (hunk 3 above)
Comment (ashokkumarkannan1) [reply python/tvm/relay/op/contrib/forge/forge_passes.py:163]: This is not required as I will be removing the asserts

**Pull request merged**